scipy/stats/tests/test_stats.py

```diff
@@ -3967,12 +3969,47 @@ def test_ttest_ind_exact_alternative(self):
         assert_equal(res_l_ab.pvalue[~mask] + res_g_ba.pvalue[~mask],
                      res_2_ab.pvalue[~mask])
 
+    def test_ttest_ind_exact_selection(self):
+        # test the various ways of activating the exact test
+        np.random.seed(0)
+        N = 3
+        a = np.random.rand(N)
+        b = np.random.rand(N)
+        res0 = stats.ttest_ind(a, b)
+        res1 = stats.ttest_ind(a, b, permutations=1000)
+        res2 = stats.ttest_ind(a, b, permutations=0)
+        res3 = stats.ttest_ind(a, b, permutations=np.inf)
+        assert(res0.pvalue != res1.pvalue)
+        assert(res2.pvalue == res1.pvalue)
+        assert(res3.pvalue == res1.pvalue)
+
+    def test_ttest_ind_exact_distribution(self):
+        # the exact distribution of the test statistic should have
+        # binom(na + nb, na) elements, all unique. This was not always true
+        # in gh-4824; fixed by gh-13661.
+        np.random.seed(0)
+        a = np.random.rand(3)
+        b = np.random.rand(4)
+
+        data = np.concatenate((a, b))
+        ma, mb = len(a), len(b)
+
+        n = 100000
+        mat_perm, permutations = _data_partitions(data, n, ma)
+
+        a = mat_perm[..., :ma]
+        b = mat_perm[..., ma:]
+        t_stat = _calc_t_stat(a, b, True)
+        assert len(set(t_stat)) == binom(ma + mb, ma)
+        assert len(t_stat) == len(set(t_stat))
```

Comment:
Could calculate `len(set(t_state))` once outside the two assertions.

Reply:
Addressed [here](https://github.com/scipy/scipy/pull/13661#discussion_r599288468); please commit if that looks better.